### DISCUSSION TITLE
Positive logFC reported where counts would suggest negative logFC?

### DISCUSSION THREAD

**Discussion by aleighbrown** [erhard-lab/grandR#20] [Q&A]
Previously for assigning log2Fold changes in total RNA for my SLAM-seq data I've been aligning data with HISAT3-N, and then running featureCounts(or running salmon on the fastqs directly for gene quantification, doesn't affect the results all that much)

However, now I'm trying out the grandR package, and I'm coming across a few descrepancies. Mostly the calls between up and downregulated are the same (differences mostly in whether a is called 'significant' in one method or the other)

e.g. these 3 genes have a positive total LFC and a total Q < 0.1 after running the `LFC()` and then `PairwiseDESeq2` functions

However, checking out the counts, either as assinged by GRAND-SLAM or with featureCounts, suggests to me that the LFC should be negative? 

![image](https://user-images.githubusercontent.com/7161524/214267915-b57c98a3-1864-40d7-af9e-306d6e8ea514.png)
![image](https://user-images.githubusercontent.com/7161524/214268072-c2bf257f-51e0-4021-9dc8-438c1d7b7264.png)

What could be the possible causes of this kind of positive LFC with even the GRAND-SLAM assigned counts coming out less? Happy to share any R data objects that might be helpful for ya'll investigations

**Comment by aleighbrown** [comment]
This data was aligned with HISAT3N and then callmd was run to fix the MD tag so that GRAND-SLAM could interpret it - furthermore I use GENCODE's GTF's and swap out the biotypes (although I doubt that is the root cause) 
```
sed -i 's/transcript_type/transcript_biotype/g' gencode_new_biotype.v34.annotation.gtf
sed -i 's/gene_type/gene_biotype/g' gencode_new_biotype.v34.annotation.gtf
```

and run GRAND-SLAM with the all genes parameter like so


```
		gedi -e Slam -plot \
		-nthreads 4 -errlm TA*1.434 \
		-genomic homo_gencode_34 -allGenes
		-prefix {params.output_folder} -progress -D\
		-reads {input} \
		-trim5p 10 -trim3p 10
```

**Comment by florianerhard** [comment]
Dear aleighbrown,
the direction of the fold changes (contrast->control or control-> contrast, as you called it) is defined by the `contrast` parameter of the `GetContrasts()` function.  If you defined something like `contrast=c("col","A","B")` the fold change A/B is computed.

Does this help? Otherwise, please provide your coldata table and how you defined contrast!
Best,
Florian

**Comment by aleighbrown** [comment]
Most the genes match, these 3 are the outliers, see table below with rows as the call by the featureCounts > deseq2 method and the columns as the call by grandR LFC

| total_rna_sig   | downregulated_gs | not_significant_gs | upregulated_gs |
|-----------------|------------------|--------------------|----------------|
| downregulated   | 763              | 5619               | 3              |   
| not_significant | 58               | 18234              | 894            |  
| upregulated     | 0                | 1150               | 102            |   


Also a few downstream targets that I know what direction they should be going are going the right way by the LFC output from grandR so i don't think I've accidentally swapped the control and contrast order

**Comment by florianerhard** [comment]
Ok, I see... that's really strange. If you send me an rds of the grandR object, we can check, without it there is no way to tell...
